File: app/src/main/java/co/id/cpn/bdmgafi/widget/RegionDropdown.java
package co.id.cpn.bdmgafi.widget;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.util.AttributeSet;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import androidx.appcompat.widget.AppCompatAutoCompleteTextView;
import java.util.ArrayList;
import java.util.List;

import co.id.cpn.bdmgafi.R;
import co.id.cpn.entity.Region;
import kotlin.jvm.internal.Intrinsics;

public final class RegionDropdown extends AppCompatAutoCompleteTextView {
    private AttributeSet attributeSet;
    private String dataSelectedId = "";
    private String dataSelectedName = "";
    private RegionDropdown instance;

    public RegionDropdown(Context context, AttributeSet attributeSet2) {
        super(context, attributeSet2);
        Intrinsics.checkNotNullParameter(context, "context");
    }

    private final RegionDropdown getInstance() {
        if (this.instance == null) {
            Context context = getContext();
            Intrinsics.checkNotNullExpressionValue(context, "context");
            this.instance = new RegionDropdown(context, this.attributeSet);
        }
        return this.instance;
    }

    public void setError(CharSequence error) {
        Drawable icon = getResources().getDrawable(R.drawable.ic_important);
        if (icon != null) {
            icon.setBounds(0, 0, icon.getIntrinsicWidth(), icon.getIntrinsicHeight());
        }
        super.setError(error, icon);
    }

    public final void setupData(List<Region> regions) {
        Intrinsics.checkNotNullParameter(regions, "regions");
        List label = new ArrayList();
        label.add("All Region");
        for (Region data : regions) {
            label.add(data.getRegionName());
        }
        setAdapter(new ArrayAdapter(getContext(), android.R.layout.simple_spinner_dropdown_item, label));
        setOnClickListener(new View.OnClickListener() {
            public final void onClick(View view) {
                RegionDropdown.m530setupData$lambda0(RegionDropdown.this, view);
            }
        });
// setOnItemClickListener(new AdapterView.OnItemClickListener(regions) {
//     public final /* synthetic * / List f$1;
//
//     {
//         this.f$1 = r2;
//     }
//
//     public final void onItemClick(AdapterView adapterView, View view, int i, long j) {
//         RegionDropdown.m531setupData$lambda1(RegionDropdown.this, this.f$1, adapterView, view, i, j);
//     }
// });
    }

    /* access modifiers changed from: private */
    /* renamed from: setupData$lambda-0  reason: not valid java name */
    public static final void m530setupData$lambda0(RegionDropdown this$0, View it) {
        Intrinsics.checkNotNullParameter(this$0, "this$0");
        this$0.showDropDown();
    }

    /* access modifiers changed from: private */
    /* renamed from: setupData$lambda-1  reason: not valid java name */
    public static final void m531setupData$lambda1(RegionDropdown this$0, List $regions, AdapterView $noName_0, View $noName_1, int position, long $noName_3) {
        if (position != 0) {
            this$0.dataSelectedName = ((Region) $regions.get(position - 1)).getRegionName();
            this$0.dataSelectedId = ((Region) $regions.get(position - 1)).getRegionSID();
            return;
        }
        this$0.dataSelectedName = "ALL";
        this$0.dataSelectedId = "ALL";
    }

    public final String getSelectedID() {
        return String.valueOf(this.dataSelectedId);
    }

    public final String getSelectedName() {
        return String.valueOf(this.dataSelectedName);
    }
}

File: app/src/main/java/co/id/cpn/bdmgafi/util_view/RegionDropdown.kt
package co.id.cpn.bdmgafi.util_view

import android.annotation.SuppressLint
import android.content.Context
import android.content.res.Resources
import android.graphics.Rect
import android.util.AttributeSet
import android.util.Log
import android.util.TypedValue
import android.widget.AdapterView
import androidx.appcompat.widget.AppCompatAutoCompleteTextView
import co.id.cpn.bdmgafi.R
import co.id.cpn.bdmgafi.ui.adapter.CheckboxDropdownAdapter
import co.id.cpn.entity.Region


class RegionDropdown @JvmOverloads constructor(
    context: Context,
    attributeSet: AttributeSet?,
) : AppCompatAutoCompleteTextView(context, attributeSet) {
    private var instance: RegionDropdown? = null
    private val attributeSet: AttributeSet? = null

    private fun getInstance(): RegionDropdown? {
        if (instance == null) {
            instance = RegionDropdown(context, attributeSet)
        }
        return instance
    }

    @SuppressLint("UseCompatLoadingForDrawables")
    override fun setError(error: CharSequence?) {
        val icon = resources.getDrawable(R.drawable.ic_important)
        icon?.setBounds(
            0, 0,
            icon.intrinsicWidth,
            icon.intrinsicHeight
        )
        super.setError(error, icon)
    }

    fun setupData(regions: List<Region>) {
        setAdapter(
            CheckboxDropdownAdapter(
                context,
                R.layout.dropdown_checkbox,
                regions
            )
        )
        this.setOnClickListener { showDropDown() }
        this.onItemClickListener = AdapterView.OnItemClickListener { parent, _, position, _ ->
            val selectedItem = parent.getItemAtPosition(position).toString()
            Log.w("SELECTED ", selectedItem)
        }
        this.setOnDismissListener {
            val regionsSelected = arrayListOf<Region>()
            regions.forEach {
                if (it.isSelected) {
                    regionsSelected.add(it)
                }
            }
            this.setText("Selected ${regionsSelected.size} items")
            dismissDropDown()
        }
    }

    override fun showDropDown() {
        val displayFrame = Rect()
        getWindowVisibleDisplayFrame(displayFrame)
        val locationOnScreen = IntArray(2)
        getLocationOnScreen(locationOnScreen)
        val bottom = locationOnScreen[1] + height
        val availableHeightBelow: Int = displayFrame.bottom - bottom
        val r: Resources = resources
        val bannerHeight = Math.round(
            TypedValue.applyDimension(
                TypedValue.COMPLEX_UNIT_DIP,
                50f,
                r.getDisplayMetrics()
            )
        )
        val downHeight = availableHeightBelow - bannerHeight
        dropDownHeight = if (downHeight > 0) {
            downHeight
        } else {
            300
        }
        super.showDropDown()
    }
}